trade-finance-manager-ui/server/controllers/case/amendments/amendCoverEndDate.test.js

```diff
@@ -39,35 +46,83 @@ describe('POST postAmendCoverEndDate', () => {
+    describe('GEF deal', () => {
+      it('should redirect to is using facility end date page when no errors', async () => {
+        api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
+        api.getFacility.mockResolvedValueOnce(gefFacility);
+        api.updateAmendment.mockResolvedValueOnce({ status: 200 });
+
+        const req = {
+          params: {
+            _id: dealId,
+            amendmentId,
+            facilityId,
+          },
+          body: {
+            'cover-end-date-day': '12',
+            'cover-end-date-month': '8',
+            'cover-end-date-year': '2030',
+          },
+          session,
+        };
+
+        await postAmendCoverEndDate(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/is-using-facility-end-date`);
+      });
+    });
 
-      api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
-      api.getFacility.mockResolvedValueOnce(facility);
-      api.updateAmendment.mockResolvedValueOnce({ status: 200 });
-
-      const req = {
-        params: {
-          _id: MOCK_AMENDMENT_COVERENDDATE_CHANGE.dealId,
-          amendmentId: MOCK_AMENDMENT_COVERENDDATE_CHANGE.amendmentId,
-          facilityId: MOCK_AMENDMENT_COVERENDDATE_CHANGE.facilityId,
-        },
-        body: {
-          'cover-end-date-day': '12',
-          'cover-end-date-month': '8',
-          'cover-end-date-year': '2030',
-        },
-        session,
-      };
-
-      await postAmendCoverEndDate(req, res);
-
-      expect(res.redirect).toHaveBeenCalledWith(
-        `/case/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.dealId}/facility/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.facilityId}/amendment/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.amendmentId}/is-using-facility-end-date`,
-      );
+    describe('BSS/EWCS deal', () => {
+      it('should redirect to the check answers page when no errors and only changing cover end date', async () => {
+        api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
+        api.getFacility.mockResolvedValueOnce(bssEwcsFacility);
+        api.updateAmendment.mockResolvedValueOnce({ status: 200 });
+
+        const req = {
+          params: {
+            _id: dealId,
+            amendmentId,
+            facilityId,
+          },
+          body: {
+            'cover-end-date-day': '12',
+            'cover-end-date-month': '8',
+            'cover-end-date-year': '2030',
+          },
+          session,
+        };
+
+        await postAmendCoverEndDate(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(`/case/${dealId}/facility/${facilityId}/amendment/${amendmentId}/check-answers`);
+      });
+
+      it('should redirect to the amend facility value page when no errors and also amending facility value', async () => {
```

Comment:
can the when no errors... go in a describe? same for other long new `it` blocks in this PR for unit tests